### PR TITLE
[CP-1831] Editing contact which is deleted via Center can still be sa…

### PR DESCRIPTION
**Pull request by lukasmas**
…ved but with incomplete data

Fix for editing contact when it was deleted via Center. Edited contact cannot be saved if it was deleted earlier. Proper communication is shown. Fix for phonebook app was not handling database notifications correctly.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

File: module-apps/application-phonebook/windows/PhonebookNewContact.cpp [hunk 1]
```diff
@@ -255,4 +260,30 @@ namespace gui
...
+        application->switchWindow(gui::window::name::dialog_confirm, std::move(metaData));
+    }
+
+    bool PhonebookNewContact::checkIfContactWasDeletedDuringEditProcess()
+    {
+        auto contactByID = DBServiceAPI::ContactGetByID(application, contact->ID);
+
+        return contactByID->size() == 1 and contactByID->front().ID == 0 and
+               contactByID->front().primaryName.empty() and contactByID->front().alternativeName.empty() and
+               contactByID->front().numbers.empty();
```
Comment (Maciej-Mudita) [inline module-apps/application-phonebook/windows/PhonebookNewContact.cpp:286]: Are these all comparisons necessary? Isn't it enough to check if `contactByID->empty()`?

File: module-apps/application-phonebook/windows/PhonebookNewContact.cpp [hunk 2]
```diff
@@ -255,4 +260,30 @@ namespace gui
...
+                this->application->switchWindow(gui::name::window::main_window, std::move(data));
+
+                return true;
+            }});
+        application->switchWindow(gui::window::name::dialog_confirm, std::move(metaData));
+    }
+
+    bool PhonebookNewContact::checkIfContactWasDeletedDuringEditProcess()
+    {
+        auto contactByID = DBServiceAPI::ContactGetByID(application, contact->ID);
```
Comment (Maciej-Mudita) [inline module-apps/application-phonebook/windows/PhonebookNewContact.cpp:282]:
```suggestion
        const auto contactByID = DBServiceAPI::ContactGetByID(application, contact->ID);
```

File: module-apps/application-phonebook/windows/PhonebookNewContact.hpp [hunk 3]
```diff
@@ -39,6 +39,8 @@ namespace gui
                                         const std::uint32_t duplicatedNumberContactID = 0u);
         void showDialogDuplicatedSpeedDialNumber();
         void setSaveButtonVisible(bool visible);
+        void showContactDeletedNotification();
+        bool checkIfContactWasDeletedDuringEditProcess();
```
Comment (Maciej-Mudita) [inline module-apps/application-phonebook/windows/PhonebookNewContact.hpp:43]:
```suggestion
        bool checkIfContactWasDeletedDuringEditProcess() const;
```

File: module-apps/application-phonebook/windows/PhonebookNewContact.cpp (hunk 1 above)
Comment (lukasmas) [reply module-apps/application-phonebook/windows/PhonebookNewContact.cpp:286]: Unfortunately not, for some reasons if the contacts do not exist in a database a vector with single empy ContactRecord is returned. Thats why I created all these checks

**Review by Maciej-Mudita — APPROVED** [review]

**Review by awulkiew — APPROVED** [review]

**Pull request merged**